fabric-loot-table-api-v2/src/main/java/net/fabricmc/fabric/api/loot/v2/LootTableLoadingCallback.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.loot.v2;
+
+import net.minecraft.loot.LootManager;
+import net.minecraft.loot.LootTable;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A callback that is called when loot tables are loaded.
+ * Use {@link #EVENT} to register instances.
+ */
+@FunctionalInterface
+public interface LootTableLoadingCallback {
+	Event<LootTableLoadingCallback> EVENT = EventFactory.createArrayBacked(
+			LootTableLoadingCallback.class,
+			(listeners) -> (resourceManager, lootManager, id, tableBuilder, setter) -> {
+				for (LootTableLoadingCallback callback : listeners) {
+					callback.onLootTableLoading(resourceManager, lootManager, id, tableBuilder, setter);
+				}
+			}
+	);
+
+	/**
+	 * Called when a loot table is loading.
+	 *
+	 * @param resourceManager the server resource manager
+	 * @param lootManager     the loot manager
+	 * @param id              the loot table ID
+	 * @param tableBuilder    a builder of the loot table being loaded
+	 * @param setter          a loot table setter for completely replacing the loaded loot table
```

Comment:
does setting the replacement cancel future notification or does it silently change return result. I imagine this could be an area of contention a mod or two may fight over.

Reply:
It silently changes the result, which is problematic IMO (and also according to @asiekierka who added it in #160). We could figure out a better solution now.

Reply:
I updated this to cancel future callbacks now. It's different behaviour than in v1's original implementation, but it's also completely undefined in v1. It doesn't make sense to continue the callback loop anyway.